src/app/ReadClient.cpp

```diff
@@ -484,6 +484,7 @@ CHIP_ERROR ReadClient::GenerateDataVersionFilterList(DataVersionFilterIBs::Build
 
 void ReadClient::OnActiveModeNotification()
 {
+    // Note: this API only works when issuing subscription via SendAutoResubscribeRequest.
```

Comment:
So hold on.  When not doing auto-resubscribe, do we want to ignore this call, or still close ourselves even though it's fatal?

Reply:
I think the whole OnActiveModeNotification is designed for auto-resubscription stuff, if they are not using OnActiveModeNotification, it is really application consumer's responsibility to decide how to manipulate close and retry logic when receiving check-in message, and application consumer need implement their CheckInDelegate and not call InteracitonModelEngine::OnActiveModeNotification.

I think we should ignore this call when not using auto-resubscription.

Reply:
just update code to reflect this change.